files/zh-cn/web/progressive_web_apps/tutorials/cycletracker/secure_connection/index.md

```diff
@@ -0,0 +1,94 @@
+
+## 使用 files:// 协议查看
+
+任何浏览器都会渲染你的 HTML。要查看你在上一节中创建的应用了 CSS 文件的 HTML 文件，可以通过你的计算机的文件结构导航到 `index.html` 文件，或者从你的浏览器使用“打开文件”菜单选项。
+
+在更新了 `index.html`，并且将 `style.css` 存放在同一目录中之后，以窄浏览器窗口查看页面时它应该看起来类似于此截图：
+
+![带有大标题，一个带有说明的表单，两个日期选择器和一个按钮的浅绿色网页。底部显示两个作用是占位符的月经周期和一个标题。](filefile.jpg)
+
+我们正在使用 `file://` 协议查看我们的页面。这适用于我们代码库的当前状态，并且在我们[添加 JavaScript 功能](/zh-CN/docs/Web/Progressive_web_apps/Tutorials/CycleTracker/JavaScript_functionality)之后将继续适用。然而，清单文件和 services worker（都是 PWA 的要求）就像许多其他 API 一样，需要一个安全连接。PWA 需要通过基于 `https` 的 web 服务器或使用 `localhost`、`127.0.0.1`（无论是否带有端口号）的本地开发环境来提供。如果我们使用 `files://` 协议查看我们完成的应用，我们的 [清单文件](/zh-CN/docs/Web/Progressive_web_apps/Tutorials/CycleTracker/Manifest_file)将被忽略，任何我们添加的 [service worker](/zh-CN/docs/Web/Progressive_web_apps/Tutorials/CycleTracker/Service_workers) 都将失败。
+
+> **备注：** 通过 `https` 提供你的应用不仅适用于 PWA，而且适用于所有网站，因为它确保了在你的 web 服务器和用户的浏览器之间传输的信息是端到端加密的。许多 [Web API 需要安全上下文](/zh-CN/docs/Web/Security/Secure_Contexts/features_restricted_to_secure_contexts)。即使你没有创建可安装的 PWA，当你向任何 web 应用添加功能时，你都可能会遇到需要安全上下文的情况。
+
+我们需要一个本地开发环境来完成教程。[使 PWA 可安装](/zh-CN/docs/Web/Progressive_web_apps/Guides/Making_PWAs_installable)的一部分是安全服务器。文件需要在 web 上通过安全连接进行服务，以获取 PWA 提供的好处，并将我们的应用程序作为 PWA 分发。
+
+## localhost
+
+根据操作系统的不同，设置本地开发环境的默认方法也不同。虽然默认的索引和配置文件的位置在不同的操作系统上可能不同，但大多数桌面操作系统都启用了你作为开发者可以接触的服务器配置。
+
+例如，在 MacOS 上，至少在 Sierra 和 Monterey 上，输入 `sudo apachectl start` 可以启用 apache 服务器。启动服务器后，在浏览器中输入 `http://localhost` 会显示一个显示“It works!”的基本网页。默认情况下，显示的 HTML 文件是 `Library/WebServer/Documents/index.html.en`。要启用除 `.html.en` 以外的文件扩展名或将根目录从 `Library/WebServer/Documents/` 更改为其他位置，你必须编辑位于 `/etc/apache2/httpd.conf` 的 apache http 配置文件。可以使用 `sudo apachectl stop` 停止服务器。
+
+操作系统默认的 `localhost` 有一个易记的 URL, 但服务器根位置难记，配置过程也很困难。它还只允许一次在一个位置使用一个本地服务器。幸运的是，有更直观的服务器设置方法，可以在多个端口上创建一个或多个本地开发环境。
+
+## 使用端口号的 localhost
+
+有几个 {{glossary("IDE")}} 扩展和特定编程语言的包可以通过单击或终端命令启动开发环境。你甚至可以启动多个本地服务器，每个使用不同的端口号。
+
+你可以使用 [VSCode 插件](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/set_up_a_local_testing_server#使用文本编辑器的拓展)运行本地 HTTP 服务器，它可以在单个或不同端口上运行本地服务器。[VSCode](https://code.visualstudio.com/download) IDE 的 [Preview on Web Server 扩展](https://marketplace.visualstudio.com/items?itemName=yuichinukiyama.vscode-preview-server)在编辑器当前打开的目录的根目录下创建一个服务器，默认端口为 `8080`。VS Code 扩展是可配置的。`previewServer.port` 设置项是 web 服务器的端口号。可以编辑和更改扩展的默认设置 `8080`。默认情况下，在浏览器的 URL 栏中输入 [`localhost:8080`](/) 将加载该页面。
+
+请注意，Preview on Web Server 扩展使用 browserSync。当你的开发环境由此扩展启动时，[`localhost:3001`](/) 提供当前服务器环境的概述。
+
+你还可以使用 IntelliJ IDE 创建 [本地服务器](https://www.jetbrains.com/help/idea/creating-local-server-configuration.html)，它搭载了一个[可配置的内置 PHP Web 服务器](https://www.jetbrains.com/help/idea/php-built-in-web-server.html#configuring-built-in-web-server)。
+
+了解如何使用 [Python](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/set_up_a_local_testing_server#使用_Python) 或本地服务器端语言（如 PHP）来[设置本地测试服务器](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/set_up_a_local_testing_server#在本地运行服务器端语言)。
```

Comment:
Please lowercase the hash. If you're not familiar with this, I recommend checking the rendered page to determine the actual value of it.

```suggestion
了解如何使用 [Python](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/set_up_a_local_testing_server#使用_python) 或本地服务器端语言（如 PHP）来[设置本地测试服务器](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/set_up_a_local_testing_server#在本地运行服务器端语言)。
```

Reply:
Apologize for my being careless again ＞﹏＜

Reply:
Thank you for your patience 👍 @yin1999